### ISSUE TITLE
Occasional "payment not found" error on `sendPayment`

### ISSUE THREAD

**Issue by ok300** [breez/breez-sdk#458]
Customer report:

> I encountered an exception while sending a payment with the `sendPayment()` method, as follows: `FfiException(RESULT_ERROR, payment not found, null).`
This is just a 2000 sats invoice generated from the Alby wallet that I'm trying to pay. The thing is, the payment did become successful, and the wallet balance did change. However, with the `await _breezBridge.sendPayment(bolt11: bolt11);` method, the exception was thrown and caught. I tried to pay another invoice to see if this particular exception is thrown again or not. Hopefully, this exception is not thrown, and the payment becomes successful, so it was just thrown once.

The payment in question:
```
{
  "bolt11": lnbc20u1pjs9l4dpp533ew4wgd8n68j30u06hdgq6fj8gfmkxqxja80n33yht5mach2x4sdq5g9kxy7fqd9h8vmmfvdjscqzzsxqyz5vqsp5ptqjepu9mthd8x0uvlx68480vhjjnrmlvrv09hn0pgwrl7dtfsgq9qyyssqdk6037rh5fq7n78cmsgy78zsnwaknqsc3heceqyz5uw6lfmehkppfp6u7k6xv0q6qz42qv3kf2mwrgvkcpmw7peacwt9vx0xlpldfhgqjal99s,
  "payeePubkey": 030a58b8653d32b99200a2334cfe913e51dc7d155aa0116c176657a4f1722677a3,
  "paymentHash": 8c72eab90d3cf47945fc7eaed4034991d09dd8c034ba77ce3125d74df71751ab,
  "description": Alby invoice,
  "descriptionHash": null,
  "amountMsat": 2000000,
  "timestamp": 1694695085,
  "expiry": 86400,
  "paymentSecret": [10, 193, 44, 135, 133, 218, 238, 211, 153, 252, 103, 205, 163, 212, 239, 101, 229, 41, 143, 127, 96, 216, 242, 222, 111, 10, 28, 63, 249, 171, 76, 16],
}
```

This is the first and only time it happened (e.g. not reproducible).

**Comment by ok300** [comment]
Maybe related to the recent #452?

**Comment by roeierez** [comment]
@dangeross I don't think it is related because the payment was succesfull.
@ok300 I suspect the lookup in the db for the payment after the sync failed, maybe because that payment wasn't retrieved from greenlight from some reason in the sync.

**Comment by dangeross** [comment]
I think it would only happen if in the sync after several retries to fetch_channels_and_balance there was no update

**Comment by roeierez** [comment]
I initially thought this way too but Carlos wrote that the balance did update...

**Comment by dangeross** [comment]
After the error though, right? I could be after the error the paid invoice came in via the stream. Which then inserts the payment, syncs, emits InvoicePaid event

**Comment by massmux** [comment]
Just to make aware, that i am experiencing the same error too. The invoice gets paid but the error appears and payment notification is not delivered.

**Comment by JssDWt** [comment]
Linking https://github.com/breez/breez-sdk/issues/291

**Comment by massmux** [comment]
I have updated yesterday the lib and frankly receive same error. i followed the links of updates but it does not seem this problem was resolved. i get this error quite each payment.

for example now paid 50 sats.
got
 breezerbot   | error paying invoice:  SdkError.SendPaymentFailed('Failed to send payment: Payment not found')

returned False

but the balance decreased 50 sats normally

unfortunately the error does not explain anything more and cannot find more info on existing docs
